app/src/main/java/xyz/klinker/blur/launcher3/QsbContainerView.java

```diff
@@ -1,270 +1,119 @@
- */
-public class QsbContainerView extends FrameLayout {
+public class QsbContainerView extends RelativeLayout implements View.OnClickListener {
+
+    private Launcher mLauncher;
 
     public QsbContainerView(Context context) {
         super(context);
     }
 
-    public QsbContainerView(Context context, AttributeSet attrs) {
-        super(context, attrs);
+    public QsbContainerView(Context context, AttributeSet attributeSet) {
+        super(context, attributeSet);
     }
 
-    public QsbContainerView(Context context, AttributeSet attrs, int defStyleAttr) {
-        super(context, attrs, defStyleAttr);
+    public void setLauncher(Launcher sLauncher) {
+        mLauncher = sLauncher;
     }
 
     @Override
-    public void setPadding(int left, int top, int right, int bottom) {
-        super.setPadding(0, 0, 0, 0);
+    public void onFinishInflate() {
+        super.onFinishInflate();
+
+         View search = findViewById(R.id.now_tab);
+         if (search != null) {
+         search.setOnClickListener(this);
+         }
+         View voice = findViewById(R.id.qsb_base);
+         if (voice != null) {
+         voice.setOnClickListener(this);
+         }
+        setOnClickListener(this);
     }
 
-    /**
-     * A fragment to display the QSB.
-     */
-    public static class QsbFragment extends Fragment implements View.OnClickListener {
-
-        private static final int REQUEST_BIND_QSB = 1;
-        private static final String QSB_WIDGET_ID = "qsb_widget_id";
-
-        private static int sSavedWidgetId = -1;
-
-        private AppWidgetProviderInfo mWidgetInfo;
-        private LauncherAppWidgetHostView mQsb;
-
-        private BroadcastReceiver mRebindReceiver = new BroadcastReceiver() {
-            @Override
-            public void onReceive(Context context, Intent intent) {
-                rebindFragment();
-            }
-        };
-
-        @Override
-        public void onCreate(Bundle savedInstanceState) {
-            super.onCreate(savedInstanceState);
-
-            IntentFilter filter = new IntentFilter(Launcher.ACTION_APPWIDGET_HOST_RESET);
-            filter.addAction(SearchManager.INTENT_GLOBAL_SEARCH_ACTIVITY_CHANGED);
-            getActivity().registerReceiver(mRebindReceiver, filter);
-        }
-
-        private FrameLayout mWrapper;
-
-        @Override
-        public View onCreateView(
-                LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-
-            if (savedInstanceState != null) {
-                sSavedWidgetId = savedInstanceState.getInt(QSB_WIDGET_ID, -1);
-            }
-            mWrapper = new FrameLayout(getActivity());
-            mWrapper.addView(createQsb(inflater, mWrapper));
-            return mWrapper;
-        }
-
-        private View createQsb(LayoutInflater inflater, ViewGroup container) {
-            if (!AppSettings.getInstance(getActivity()).showSearchBar) {
```

Comment:
This is where the `showSearchBar` setting is. We should be able to move this to the `onFinishInflate` method and just set the visibility of this `View` to `GONE` inside there.